core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -710,6 +710,11 @@ class SparkContext(config: SparkConf) extends Logging {
     }
   }
 
+  /** Set spark conf */
+  def setSparkConf(sparkConf: SparkConf): Unit = {
```

Comment:
Spark cknfig is supposed to be immutable. I don't think we should allow this in Spark context. If you should set a static SQL config, you should either stop and start the context again or set it initially when you create a Spark context.

Reply:
Agree with @HyukjinKwon. Static SQL configs are static and immutable. Forcefully modifying the configs could cause unexpected behavior.